internal/elasticsearch/index/commons.go

```diff
@@ -80,3 +80,30 @@ func FlattenIndexAlias(name string, alias models.IndexAlias) (interface{}, diag.
 
 	return a, diags
 }
+
+func ExpandLifecycle(definedLifecycle *schema.Set) (*models.LifecycleSettings, diag.Diagnostics) {
```

Comment:
```suggestion
func ExpandLifecycle(definedLifecycle *schema.Set) *models.LifecycleSettings {
```

There's never any actual diags returned here.